cmd/backport-verifier/main.go

```diff
@@ -0,0 +1,185 @@
+			return fmt.Errorf("%s should be in org/repo format", upstreamRepo)
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+type options struct {
+	mut *sync.RWMutex
+
+	configPath        string
+	webhookSecretFile string
+
+	config *Config
+
+	githubEventServerOptions githubeventserver.Options
+	github                   prowflagutil.GitHubOptions
+
+	dryRun bool
+}
+
+func gatherOptions() options {
+	o := options{
+		mut: &sync.RWMutex{},
+	}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.configPath, "config-path", "", "Path to publicize configuration.")
+	fs.StringVar(&o.webhookSecretFile, "hmac-secret-file", "", "Path to the file containing the GitHub HMAC secret.")
+
+	o.github.AddFlags(fs)
+	o.githubEventServerOptions.Bind(fs)
+
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatalf("cannot parse args: '%s'", os.Args[1:])
+	}
+	return o
+}
+
+func (o *options) Validate() error {
+	if err := o.github.Validate(o.dryRun); err != nil {
+		return err
+	}
+
+	bytes, err := ioutil.ReadFile(o.configPath)
+	if err != nil {
+		return fmt.Errorf("couldn't read publicize configuration file: %v", o.configPath)
+	}
+
+	if err := yaml.Unmarshal(bytes, &o.config); err != nil {
+		return fmt.Errorf("couldn't unmarshal configuration: %w", err)
+	}
+
+	if err := o.config.validate(); err != nil {
+		return err
+	}
+
+	if err := o.githubEventServerOptions.DefaultAndValidate(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *options) getConfigWatchAndUpdate() (func(ctx context.Context), error) {
+	errFunc := func(err error, msg string) {
+		logrus.WithError(err).Error(msg)
+	}
+
+	eventFunc := func() error {
+		bytes, err := ioutil.ReadFile(o.configPath)
+		if err != nil {
+			return fmt.Errorf("couldn't read configuration file %s: %w", o.configPath, err)
+		}
+
+		var c *Config
+		if err := yaml.Unmarshal(bytes, &c); err != nil {
+			return fmt.Errorf("couldn't unmarshal configuration: %w", err)
+		}
+
+		if err := c.validate(); err != nil {
+			return err
+		}
+
+		o.mut.Lock()
+		defer o.mut.Unlock()
+		o.config = c
+		logrus.Info("Configuration updated")
+
+		return nil
+	}
+	watcher, err := prowconfig.GetCMMountWatcher(eventFunc, errFunc, filepath.Dir(o.configPath))
+	if err != nil {
+		return nil, fmt.Errorf("couldn't get the file watcher: %w", err)
+	}
+
+	return watcher, nil
+}
+
+func main() {
+	logrusutil.ComponentInit()
+
+	o := gatherOptions()
+	if err := o.Validate(); err != nil {
+		logrus.Fatalf("Invalid options: %v", err)
+	}
+
+	configWatchAndUpdate, err := o.getConfigWatchAndUpdate()
+	if err != nil {
+		logrus.WithError(err).Fatal("couldn't get config file watch and update function")
+	}
+	interrupts.Run(configWatchAndUpdate)
+
+	secretAgent := &secret.Agent{}
+	if err := secretAgent.Start([]string{o.github.TokenPath, o.webhookSecretFile}); err != nil {
+		logrus.WithError(err).Fatal("Error starting secrets agent.")
+	}
+
+	logger := logrus.NewEntry(logrus.New())
```

Comment:
I believe this short-circuits the initialization in `ComponentInit` as that only applies to the `StandardLogger` or not?

Reply:
It should be global - but what's a better way?